bot/moderation/moderation.py

```diff
@@ -22,17 +26,540 @@ def __init__(self, bot):
+        """
+        self._db_connector.remove_member_warnings(user.id)
+        await ctx.send(f"Alle Verwarnungen für {user.mention} wurden erfolgreich aufgehoben. :white_check_mark:")
+
+    @commands.command(name='mute', hidden=True)
+    @command_log
+    async def mute_user(self, ctx: commands.Context, user: discord.Member, *, reason: Optional[str]):
+        """Command Handler for the `mute` command.
+
+        Mutes the specified member server-wide. This means the user won't be able to write messages, join voice channels
+        or even add reactions. This is done by granting the configured mute role.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            user (discord.Member): The member who should be muted.
+        """
+        if self.role_muted in user.roles:
+            await ctx.send("Dieser Nutzer ist bereits stummgeschalten. :flushed:")
+            return
+
+        await user.add_roles(self.role_muted, reason=reason)
+        await ctx.send(f"{user.mention} wurde stummgeschalten. :mute:")
+
+    @commands.command(name='unmute', hidden=True)
+    @command_log
+    async def unmute_user(self, ctx: commands.Context, user: discord.Member):
+        """Command Handler for the `unmute` command.
+
+        Unmutes the specified member on the server. This is done by removing the configured mute role from him.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            user (discord.Member): The member who should be unmuted.
+        """
+        if self.role_muted not in user.roles:
+            await ctx.send("Dieser Nutzer ist nicht stummgeschalten. :thinking:")
+            return
+
+        await user.remove_roles(self.role_muted)
+        await ctx.send(f"{user.mention} ist nicht mehr stummgeschalten. :speaker:")
+        await user.send(f"Hey, {user.display_name}! :wave:\nDu bist nicht mehr stummgeschalten! :speaker: Versuch "
+                        f"bitte in Zukunft, dich mehr an unsere {self.ch_rules.mention} zu halten, da wir ansonsten "
+                        f"gezwungen sind, härtere Strafen zu verhängen. :scales:")
+
+    @commands.command(name='tempmute', hidden=True)
+    @command_log
+    async def tempmute_user(self, ctx: commands.Context, user: discord.Member, duration: str, *, reason: Optional[str]):
+        """Command Handler for the `tempmute` command.
+
+        Temporarily mutes the specified member server-wide for the duration given. This means the user won't be able to
+        write messages, join voice channels or even add reactions. This is done by granting the configured mute role.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            user (discord.Member): The member who should be muted.
+            duration (str): The amount of time the user should be banned from the server.
```

Comment:
Format of duration?

Reply:
Will add this link as a comment: https://github.com/wroberts/pytimeparse